modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/AudioProcessorImpl.java

```diff
@@ -42,14 +43,18 @@
 import java.util.Set;
 import java.util.Vector;
 
-import org.apache.commons.lang.time.StopWatch;
+import org.osgi.service.component.annotations.Component;
 
 /**
  * @author Juan González
  * @author Sergio González
  * @author Mika Koivisto
  * @author Ivica Cardic
  */
+@Component(
+	property = "type=" + DLProcessorConstants.AUDIO_PROCESSOR,
+	service = {AudioProcessor.class, DLProcessor.class}
```

Comment:
@MrJohn1911 @rafaprax Why define two types of service here? I have the same question at https://github.com/liferay-core-infra/liferay-portal/pull/3234#issuecomment-1419649717

Reply:
Hi @tinatian I already answered you [here](https://github.com/liferay-core-infra/liferay-portal/pull/3392#issuecomment-1422868834)